### ISSUE TITLE
Browser showing smeared thumbnails

### ISSUE THREAD

**Issue by jlaura** [radiantearth/stac-browser#369]
See: https://stac.astrogeology.usgs.gov/browser-dev/#/api/collections/lunar_orbiter_laser_altimeter?.language=en

<img width="733" alt="Screenshot 2023-09-12 at 3 17 59 PM" src="https://github.com/radiantearth/stac-browser/assets/2105468/ffdc3bb4-9171-46a7-857f-19476a24fd5c">

Is this perhaps because the projection is being described at the item level and the browser is assuming that the projection information also applies to the thumbnail? Or is something else happening causing the thumbnails to display at some upper left origin and then smear across the context map?

**Comment by DanielJDufour** [comment]
Hi, @jlaura . Just wanted to let you know that I saw this.  If I had to guess, I'd say it's an issue with georaster-layer-for-leaflet stumbling on the custom projection or a custom geotransform.  I don't think we are currently trying to reproject thumbnails.  I'll try to look into it within a fortnight. I'm sorry for the inconvenience this is causing.  Looks like a really cool dataset!

**Comment by DanielJDufour** [comment]
I looked into this a little more.  I think my assumption that the issue was with georaster-layer-for-leaflet is incorrect for this specific example.  It looks like the issues appear in the overlay pane, which is consistent with displaying ImageOverlay [layers](https://leafletjs.com/reference.html#imageoverlay).  The thumbnails appear almost square-like, but for some reason Leaflet is rendering them as very non-square rectangles.  My best guess is that the code is tripping up over the projection or miscalculating the bounds of the thumbnail some how.  Maybe the answer can be found here? https://github.com/stac-utils/stac-layer/blob/main/src/add.js#L178

In either case, very interesting problem indeed.  Thank you for bringing this to our attention and for providing such an awesome use case!

<img width="1439" alt="Screenshot 2023-09-24 at 12 51 15 PM" src="https://github.com/radiantearth/stac-browser/assets/4313463/a2eba2bd-6254-4ff5-97a6-3c10f8e1bd7b">

**Comment by DanielJDufour** [comment]
Tagging @m-mohr , who might have an idea!

**Comment by m-mohr** [comment]
Will check in October once I'm back at work.

**Comment by m-mohr** [comment]
The issue likely comes from https://github.com/stac-utils/stac-layer/blob/v0.15.0/src/utils/bboxToLatLngBounds.js

There it expects the bbox to consist of 4 numbers, but here we get 6 (includes z). That will likely make the bounds invalid and as such they get smeared. I think it should be fixed by the new codebase, but it hasn't be integrated in STAC Browser yet.

The least time consuming option to solve this would be to release v0.15.1 with a fix, but I'm not sure whether anyone has time/funding to work on this right now?

**Comment by m-mohr** [comment]
Indeed, if I fix this function, the images don't get distorted (but in 1.0.0 they get shown too large, so there's more work to do).
I don't have enough time right now to dig deeper unfortunately.

**Comment by m-mohr** [comment]
I can't reproduce this anylonger in v3.3.0.
